frontend/kesaseteli/employer/src/components/application/steps/step1/Step1Employer.tsx

```diff
@@ -0,0 +1,54 @@
+import CompanyInfo from 'kesaseteli/employer/components/application/companyInfo/CompanyInfo';
+import ActionButtons from 'kesaseteli/employer/components/application/form/ActionButtons';
+import TextInput from 'kesaseteli/employer/components/application/form/TextInput';
+import StepForm from 'kesaseteli/employer/components/application/StepForm';
+import useApplicationApi from 'kesaseteli/employer/hooks/application/useApplicationApi';
+import { useTranslation } from 'next-i18next';
+import React from 'react';
+import FormSection from 'shared/components/forms/section/FormSection';
+import EmployerApplication from 'shared/types/employer-application';
+
+import { $EmployerBasicInfo } from './Step1Employer.sc';
+
+const Step1Employer: React.FC = () => {
+  const { t } = useTranslation();
+  const { updateApplication } = useApplicationApi();
+
+  const onSubmit = (draftApplication: EmployerApplication): void => {
+    updateApplication(draftApplication);
+  };
+  const stepTitle = t('common:application.step1.header');
+  return (
+    <StepForm stepTitle={stepTitle}>
+      <FormSection
+        header={stepTitle}
+        tooltip={t('common:application.step1.tooltip')}
+      >
+        <CompanyInfo />
+      </FormSection>
+      <FormSection>
+        <$EmployerBasicInfo>
+          <TextInput
+            id="invoicer_name"
+            validation={{ required: true, maxLength: 256 }}
+          />
+          <TextInput
+            id="invoicer_email"
+            validation={{
+              required: true,
+              maxLength: 254, // eslint-disable-next-line security/detect-unsafe-regex
+              pattern: /^(([^\s"(),.:;<>@[\\\]]+(\.[^\s"(),.:;<>@[\\\]]+)*)|(".+"))@((\[(?:\d{1,3}\.){3}\d{1,3}])|(([\dA-Za-z-]+\.)+[A-Za-z]{2,}))$/,
+            }}
+          />
+          <TextInput
+            id="invoicer_phone_number"
+            validation={{ required: true, maxLength: 64 }}
+          />
```

Comment:
Having the validation inline here is error-prone, and makes testing hard.

Reply:
There is already validation tests in application.test.tsx [here](https://github.com/City-of-Helsinki/yjdh/pull/204/files#diff-6c92d4860f92954993b55ba77efb5f9062292779297df9293651056f441cd6beR102) and [here](https://github.com/City-of-Helsinki/yjdh/pull/204/files#diff-6c92d4860f92954993b55ba77efb5f9062292779297df9293651056f441cd6beR78)

if you mean unit tests, they should be written for email regex and other upcoming regexes. In kesaseteli project the validations will be much more relaxed compared to benefit's so mostly min and max lengths only which are determined by database.
I'm planning to add email and other regex unit tests soon when there will be more validation related stuff coming with step 2.